src/main/java/software/amazon/event/kafkaconnector/EventBridgeWriter.java

```diff
@@ -157,173 +169,55 @@ public List<EventBridgeWriterRecord> putItems(List<EventBridgeWriterRecord> reco
-          record.setError(errorCode, errorMessage);
-        });
-  }
-
-  /**
-   * This method serializes the Kafka record to JSON format, which is the required format for
-   * EventBridge.
-   *
-   * @param record Kafka record to be sent to EventBridge
-   * @return string representation of the event sent to EventBridge
-   */
-  private String createJsonPayload(SinkRecord record) {
-    try {
-      var root = objectMapper.createObjectNode();
-      root.put("topic", record.topic());
-      root.put("partition", record.kafkaPartition());
-      root.put("offset", record.kafkaOffset());
-      root.put("timestamp", record.timestamp());
-      root.put("timestampType", record.timestampType().toString());
-      root.putIfAbsent("headers", createHeaderArray(record));
-
-      if (record.key() == null) {
-        root.putIfAbsent("key", null);
-      } else {
-        root.putIfAbsent(
-            "key",
-            createJSONFromByteArray(
-                jsonConverter.fromConnectData(record.topic(), record.keySchema(), record.key())));
-      }
-
-      // tombstone handling
-      if (record.value() == null) {
-        root.putIfAbsent("value", null);
-      } else {
-        root.putIfAbsent(
-            "value",
-            createJSONFromByteArray(
-                jsonConverter.fromConnectData(
-                    record.topic(), record.valueSchema(), record.value())));
-      }
-
-      return root.toString();
-    } catch (Exception e) {
-      throw new DataException("Record can not be converted to json", e);
-    }
-  }
-
-  /**
-   * This method serializes Kafka message headers to JSON.
-   *
-   * @param record Kafka record to be sent to EventBridge
-   * @return headers to be added to EventBridge message
-   * @throws JsonProcessingException
-   */
-  private ArrayNode createHeaderArray(SinkRecord record) throws JsonProcessingException {
-    var headersArray = objectMapper.createArrayNode();
-
-    for (Header header : record.headers()) {
-      var headerItem = objectMapper.createObjectNode();
-      headerItem.putIfAbsent(
-          header.key(),
-          createJSONFromByteArray(
-              jsonConverter.fromConnectHeader(
-                  record.topic(), header.key(), header.schema(), header.value())));
-      headersArray.add(headerItem);
-    }
-    return headersArray;
+  private Stream<EventBridgeResult<EventBridgeEventId>> mapResponseToResult(
+      List<MappedSinkRecord<PutEventsRequestEntry>> request, PutEventsResponse response) {
+    return IntStream.range(0, request.size())
+        .mapToObj(
+            index -> {
+              var sinkRecord = request.get(index).getSinkRecord();
+              var resultEntry = response.entries().get(index);
+
+              if (resultEntry.eventId() == null) {
+                var code = resultEntry.errorCode();
+                var message = resultEntry.errorMessage();
+                // TODO remove, since error handler should log
```

Comment:
Do we need those TODOs/are they still accurate?

Reply:
It's a question. I would remove the log message since it is logged and/or send to DLQ.

Reply:
@embano1 ?